### PR TITLE
[unify] Add pack version

### PR DESCRIPTION
**Pull request by shmuel44**
## Related Issues
fixes: [link](https://jira-dc.paloaltonetworks.com/browse/CIAC-10484)
content [PR](https://github.com/demisto/content/pull/35449) link

### CODE REVIEW

File: demisto_sdk/commands/prepare_content/integration_script_unifier.py [hunk 1]
```diff
@@ -491,12 +491,19 @@ def insert_pack_version(
...
             )
-        elif script_type in {".py", ".ps1"}:
+        elif script_type == ".ps1":
             return (
                 script_code
                 if "### pack version:" in script_code
                 else f"### pack version: {pack_version}\n{script_code}"
             )
+        elif script_type == ".py":
+            if "### pack version:" not in script_code:
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/integration_script_unifier.py:501]: if it is already in the code, I think we should raise an error, @dantavori wdyt? 
If we don't, we may end up having multiple assignments to the env var, and only the latest will be used. 

(this also means we should remove it from the code when accepting contributions, I think we already have some code that removes some similar prefix)

File: demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py [hunk 2]
```diff
@@ -564,30 +564,46 @@ def mocked_get_api_module_code(*args, **kwargs):
...
-    assert version_str not in IntegrationScriptUnifier.insert_pack_version(
-        ".js", DUMMY_SCRIPT, "1.0.3"
-    )
-    version_str_js = "// pack version: 1.0.3"
-    assert version_str_js not in DUMMY_SCRIPT
-    assert version_str_js in IntegrationScriptUnifier.insert_pack_version(
-        ".js", DUMMY_SCRIPT, "1.0.3"
-    )
+    pack_version_1 = "1.0.3"
+    pack_version_2 = "1.0.4"
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py:580]: split pack_1 and pack_2 to two tests (or parametrize)

File: demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py [hunk 3]
```diff
@@ -564,30 +564,46 @@ def mocked_get_api_module_code(*args, **kwargs):
...
+    version_str_py = f"demisto.debug('pack version = {pack_version_1}')"
+    version_str_py_2 = f"demisto.debug('pack version = {pack_version_2}')"
+    version_str_ps1 = f"### pack version: {pack_version_1}"
+    version_str_js = f"logDebug('pack version = {pack_version_1}')"
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py:584]: remove js

File: .changelog/4437.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Added to the ***unify*** command that appends to content-item of type Python, logging with the pack version.
```
Comment (dorschw) [inline .changelog/4437.yml:2]:
```suggestion
- description: The ***unify*** command now logs the pack name and version in Python files.
```

File: demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py [hunk 5]
```diff
@@ -561,33 +561,77 @@ def mocked_get_api_module_code(*args, **kwargs):
...
-    version_str = "### pack version: 1.0.3"
-    assert version_str not in DUMMY_SCRIPT
-    assert version_str in IntegrationScriptUnifier.insert_pack_version(
-        ".py", DUMMY_SCRIPT, "1.0.3"
-    )
-    assert version_str in IntegrationScriptUnifier.insert_pack_version(
-        ".ps1", DUMMY_SCRIPT, "1.0.3"
+    pack_version = "1.0.3"
+    version_str_ps1 = f"### pack version: {pack_version}"
+    version_str_js = f"// pack version: {pack_version}"
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py:579]: `assert f"### pack version: {pack_version}" in updated_script_js` is clearer imo

File: demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py [hunk 6]
```diff
@@ -561,33 +561,77 @@ def mocked_get_api_module_code(*args, **kwargs):
...
+            id="script without version",
+        ),
+        pytest.param(
+            """
+            demisto.debug('pack version = 1.0.3')
+            def main():
+            """,
+            "1.0.4",
+            "demisto.debug('pack version = 1.0.4')",
+            id="script with version",
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py:616]: twice?

File: demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py [hunk 7]
```diff
@@ -561,33 +561,77 @@ def mocked_get_api_module_code(*args, **kwargs):
...
+):
+    """
+    Test that the pack version is correctly inserted into the Python script.
+    """
+    updated_script = IntegrationScriptUnifier.insert_pack_version(
+        ".py", dummy_script, pack_version
     )
+    assert (
+        expected_version_str in updated_script
+    ), "Pack version string not found in the updated script"
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py:631]: redundant, the next line contains this one

File: demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py [hunk 8]
```diff
@@ -561,33 +561,77 @@ def mocked_get_api_module_code(*args, **kwargs):
...
+    assert (
+        updated_script.count(expected_version_str) == 1
+    ), "Pack version string found more than once in the updated script"
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py:634]: can do something like `pack version found {...} times in updated script, expected only once`

File: demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py (hunk 6 above)
Comment (shmuel44) [reply demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py:616]: ?

File: .changelog/4437.yml [hunk 9]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: The ***unify*** command now logs the pack name and version in Python files.
```
Comment (dorschw) [inline .changelog/4437.yml:2]:
```suggestion
- description: The ***unify*** command now logs the pack name and version in Python files.
```

doesn't it happen (also?) in `prepare-content`?

Comment (dorschw) [inline .changelog/4437.yml:2]:
```suggestion
- description: The ***unify*** command now adds a `demisto.debug` logging of the pack name and version, in Python files.
```

**Review by dorschw — APPROVED** [review]

File: .changelog/4437.yml (hunk 9 above)
Comment (shmuel44) [reply .changelog/4437.yml:2]: I made changes to the "unify" command, probably in the "prepare-content" we are calling to the "unify" command.

Comment (dorschw) [reply .changelog/4437.yml]: I think it's worth mentioning both

**Pull request merged**